source/extensions/priv/server/ntds_decrypt.c

```diff
@@ -0,0 +1,194 @@
+/*!
+* @file ntds_decrypt.c
+* @brief Definitions for NTDS decryption functions
+*/
+#include "precomp.h"
+
+/*!
+* @brief Convert bytes into a Hex string representing those bytes.
+* @param data Pointer to the byte array we are converting
+* @param length Integer representing the length of the byte array
+* @param output Pointer to the string we are outputting the result to
+*/
+void bytes_to_string(LPBYTE data, int length, LPSTR output){

```

Comment:
Since lengths shouldn't store negative lengths I guess you can use unsigned here